src/maths_utils.c

```diff
@@ -74,3 +74,21 @@ uint8_t ulog2(uint32_t value)
 	return (sizeof(uint8_t) * 8U) - result;
 #endif
 }
+
+uint32_t swd_odd_parity(const uint32_t value)
+{
+#if defined(__GNUC__)
+	/* Ask for the libgcc impl */
+	return __builtin_parity(value);
+#elif defined(_MSC_VER)
+	/* Ask for a CPU insn */
+	return __popcount(value) & 1U;
+#else
+	/* Generic impl */
+	uint32_t result = 0;
+	for (uint32_t bit = 0; bit < 32; ++bit)
+		result ^= value & (1U << bit);
```

Comment:
This needs to be `result ^= (value >> bit) & 1U;`, otherwise it expresses a very slow bitwise copy of `value` to `result` as `^=` is bitwise, not logical.

Reply:
Hate bitwise loops. Fixed.